scipy/linalg/tests/test_lapack.py

```diff
@@ -2357,28 +2324,22 @@ def test_gtsvx_error_singular(dtype, trans, fact):
     # generate random solution x
     x = generate_random_dtype_array((n, 2), dtype=dtype)
     # create b from x for equation Ax=b
-    if trans == 'N':
-        b = A @ x
-    else:
-        b = (A.conj().T) @ x
+    trans = "T" if dtype in REAL_DTYPES else "C"
+    b = (A.conj().T if trans_bool else A) @ x
 
     # set these to None if fact = 'N', or the output of gttrf is fact = 'F'
-    dlf_, df_, duf_, du2f_, ipiv_, info_ = gtsvx_factorize_gttrf(fact,
-                                                                 (dl, d, du))
+    dlf_, df_, duf_, du2f_, ipiv_, info_ = \
+        gttrf(dl, d, du) if fact == 'F' else [None]*6
 
     gtsvx_out = gtsvx(dl, d, du, b, fact=fact, trans=trans, dlf=dlf_, df=df_,
                       duf=duf_, du2=du2f_, ipiv=ipiv_)
     dlf, df, duf, du2f, ipiv, x_soln, rcond, ferr, berr, info = gtsvx_out
     # test with singular matrix
     # no need to test inputs with fact "F" since ?gttrf already does.
     if fact == "N":
-
-        L, U = gttrf_to_lu(n, dtype, dlf, df, duf, du2f, ipiv)
-        # check that the outputs define an LU decomposition of A
-        assert_allclose(A, L @ U, atol=atol)
```

Comment:
Why not test this? We thought we should check that the output is the  LU factorization as it should be.

Reply:
This is already being tested in gttrf/gttrs it should fail there and not here.

Reply:
I agree that if the Fortran routine were not working properly, the test would fail in gttrf/gttrf. But we're not really testing the Fortran routines so much as the wrappers, right? With the changes, the tests would now pass regardless of whether `dlf` through `ipiv` were wrapped correctly, right? Even if `?gtsvx` is working properly in Fortran, the data coming out of the Python wrapper could be garbage, and it wouldn't be caught by the current tests.

Reply:
I agree but that particular part has no relation with the gtsvx per se. I know you want extra guarantee that such gttrf based check is done. But I think that connection is not there because it we don't use the gttrf based information tied to gtsvx stuff. Or maybe I am not seeing it. As is, it looked to me like a separate gttrf test. But I don't have any strong preference we can put it back in.

Reply:
@Kai-Striega Can we get a third opinion? If you think it can be left out, would you merge? If you think it should be added back in, would you do that?

Reply:
I think leaving it out is OK. At the end of the day we're testing the behaviour of gtsvx. If one of the intermediate functions fails then it should be covered by that function's tests.